pkg/languagedetection/detector.go

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package languagedetection
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type LanguageName string
+
+var (
+	python  LanguageName = "python"
+	java    LanguageName = "java"
+	unknown LanguageName = ""
+)
+
+type Language struct {
+	Name LanguageName
+}
+
+// knownPrefixes maps languages names to their prefix
+var knownPrefixes = map[string]LanguageName{
+	"python": python,
+	"java":   java,
+}
+
+func languageNameFromCommandLine(cmdline []string) (LanguageName, error) {
+	exe := getExe(cmdline)
+	for prefix, language := range knownPrefixes {
+		if strings.HasPrefix(exe, prefix) {
```

Comment:
a problem we had is that we needed to distinguish between `java` and `javac`, here we'll report `javac` as Java.
suggesting to get the "validator" function as an argument, so we can control the precision

Reply:
I think that's a good idea. I will add it